examples/customer-managed-key/main.tf

```diff
@@ -1,6 +1,10 @@
 terraform {
   required_version = ">= 1.0.0"
   required_providers {
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = ">= 2.15.0"
```

Comment:
version constraint missing see https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---providers-must-be-declared-in-the-required_providers-block-in-terraformtf-and-must-have-a-constraint-on-minimum-and-maximum-major-version

Reply:
As per [TFNFR36](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block) set `prevent_deletion_if_contains_resources` in `providers`:
```terraform
resource_group {
      prevent_deletion_if_contains_resources = true
    }
```